### PR TITLE
Provide pre-computed max sequence to remove unnecessary kernels and D2H copies

### PR DESCRIPTION
**Pull request by erhoo82**
Pass pre-computed max query and key-value sequence of a batch of sequences. This will remove some compute kernels and importantly D2H copies thus avoiding H2D syncs.

### CODE REVIEW

File: transformer_engine/pytorch/attention.py [hunk 1]
```diff
@@ -2027,6 +2037,12 @@ def forward(
         cu_seqlens_kv: Optional[torch.Tensor], default = `None`
                    Cumulative sum of sequence lengths in a batch for `key_layer` and `value_layer`,
                    with shape [batch_size + 1] and dtype torch.int32.
+        max_seqlen_q: Optional[int], default = `None`
+                   Max sequence length of the batch of query sequences.
+                   Calculate when not provided.
+        max_seqlen_kv: Optional[int], default = `None`
+                   Max sequence length of the batch of key-value sequences.
+                   Calculate when not provided.
```
Comment (timmoon10) [inline transformer_engine/pytorch/attention.py:2045]:
```suggestion
        max_seqlen_q: Optional[int], default = `None`
                      Maximum sequence length in `query_layer`.
                      Calculated from `cu_seqlens_q` if not provided.
        max_seqlen_kv: Optional[int], default = `None`
                       Maximum sequence length in `key_layer` and `value_layer`.
                       Calculated from `cu_seqlens_kv` if not provided.
```

**Review by timmoon10 — APPROVED** [review]
LGTM

**Comment by erhoo82** [comment]
Please help merge this PR.

**Pull request merged**